### ISSUE TITLE
Custom multi network model

### ISSUE THREAD

**Issue by vimmoos** [skorch-dev/skorch#832]
I'm using Pytorch for an Adversarial Variation Bayes, which is a form
of adversarial autoencoder. 
I'm trying to conform my model to skorch but i'm encountering few
problems regarding: 

1. Parameters extraction from the network for multiple optimizers 
   The parameters of the torch module needs to be divided for N
   optimizers (in my case I have 2 of them)  
2. Multiple loss
   I have multiple forward output on which i need to call the
   backwards method (indipendently and with different parameters)

A dummy example can be:
```python
class MyModule(nn.Module):

    def __init__(self, num_units):
        super().__init__()
        self.disc_dense = nn.Linear(20, num_units)
        self.ae_dense = nn.Linear(20, num_units)

    def forward(self, X, **kwargs):
        return {
            "ae": F.softmax(self.dense(X)),
            "disc": F.softmax(self.dense(X)),
        }


class Loss(nn.Module):

    def __init__(self):
        super().__init__()
        self.mse = nn.MSELoss()

    def forward(self, ae, disc, x):
        return {"disc": self.mse(disc, x), "ae": self.mse(ae, x)}
```

So in this example what i'm trying  to achive is that,First i initialize
the two optimizers with the two dense layers. Let's say that i what all
the network named_parameters (not hyperparameters) that starts with "ae" to be used in optimizer0
and all the named_parameters that start with "disc" to be used in
optimizer1.
My question is what methods should i override from the NeuralNet
class in order to achive this behaviour without disrupting the logic
of the class?
From what i see i can override get_loss for unpacking the forward
output, train_step_single for the backward call. But what i do not
understand  what method should i override to achive the splitting of
the different optimizer with different parameters.

**Comment by BenjaminBossan** [comment]
> what i do not
> understand what method should i override to achive the splitting of
> the different optimizer with different parameters.

Could you take a look at the [section on customization](https://skorch.readthedocs.io/en/latest/user/customization.html#initialization-and-custom-modules)? It shows an example with 2 modules and 2 optimizers. So it's not exactly the same as you presented, but still looks relevant.

**Comment by BenjaminBossan** [comment]
@vimmoos Did this solve your problems?

**Comment by vimmoos** [comment]
@BenjaminBossan  yeah indeed. it was exactly the example i was looking for. Modify the initiliaze_* functions was the solution  i was looking for.  I also needed to modify the train_step_single because i wanted to call the backward in a specific way (i.e. with retain_graph=True). Thank you for the ready reply.
